# BADM Project


### Dataset Info: https://www.kaggle.com/uciml/autompg-dataset

* Jeet Mav

In [1]:
#Importing Pandas to read dataset as a dataframe
import pandas as pd
import numpy as np

In [2]:
np.random.seed(25)

In [3]:
#Importing the dataset
df = pd.read_csv('auto-mpg.csv')

In [4]:
#Dataset info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    398 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   car name      398 non-null    object 
dtypes: float64(3), int64(4), object(2)
memory usage: 28.1+ KB


In [5]:
#Reading 10 Random samples from the Dataset
df.sample(10)

mpg  cylinders  displacement horsepower  weight  acceleration  \
121  15.0          8         318.0        150    3399          11.0   
162  15.0          6         258.0        110    3730          19.0   
326  43.4          4          90.0         48    2335          23.7   
200  18.0          6         250.0         78    3574          21.0   
263  17.7          6         231.0        165    3445          13.4   
285  17.0          8         305.0        130    3840          15.4   
297  25.4          5         183.0         77    3530          20.1   
385  38.0          4          91.0         67    1995          16.2   
379  36.0          4          98.0         70    2125          17.3   
23   26.0          4         121.0        113    2234          12.5   

     model year  origin                         car name  
121          73       1                dodge dart custom  
162          75       1                      amc matador  
326          80       2               vw dasher (diesel)  
200          76       1                ford granada ghia  
263          78       1  buick regal sport coupe (turbo)  
285          79       1        chevrolet caprice classic  
297          79       2               mercedes benz 300d  
385          82       3                    datsun 310 gx  
379          82       1                   mercury lynx l  
23           70       2                         bmw 2002

### Dropping the unrequired columns "car name" & "origin"

In [6]:
#Dropping the columns 'car name' and 'origin'
columns = ['car name', 'origin']
df.drop(columns, axis = 1,inplace = True )

In [7]:
#Printing random 10 samples from the dataframe
df.sample(10)

mpg  cylinders  displacement horsepower  weight  acceleration  \
126  21.0          6         200.0          ?    2875          17.0   
233  29.0          4          97.0         78    1940          14.5   
263  17.7          6         231.0        165    3445          13.4   
246  32.8          4          78.0         52    1985          19.4   
51   30.0          4          79.0         70    2074          19.5   
232  16.0          8         351.0        149    4335          14.5   
230  15.5          8         350.0        170    4165          11.4   
194  22.5          6         232.0         90    3085          17.6   
327  36.4          5         121.0         67    2950          19.9   
103  11.0          8         400.0        150    4997          14.0   

     model year  
126          74  
233          77  
263          78  
246          78  
51           71  
232          77  
230          77  
194          76  
327          80  
103          73

In [8]:
#Checking rows with ? in horsepower
df[df['horsepower'] == '?']

mpg  cylinders  displacement horsepower  weight  acceleration  \
32   25.0          4          98.0          ?    2046          19.0   
126  21.0          6         200.0          ?    2875          17.0   
330  40.9          4          85.0          ?    1835          17.3   
336  23.6          4         140.0          ?    2905          14.3   
354  34.5          4         100.0          ?    2320          15.8   
374  23.0          4         151.0          ?    3035          20.5   

     model year  
32           71  
126          74  
330          80  
336          80  
354          81  
374          82

### As we can see there are some "?" in Horsepower column we will be dropping these rows

In [9]:
# Creating a new dataframe where horsepower is not equal to "?"
df_new = df[df['horsepower'] != '?'] 

# Print the new dataframe shape
df_new.shape 


(392, 7)

### We have a total of 392 records (rows) and 7 properties (columns)

In [10]:
df_new['horsepower'] = df_new['horsepower'].astype('int64')

<ipython-input-10-d68539b1b7bc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new['horsepower'] = df_new['horsepower'].astype('int64')


In [11]:
#Checking for the type of horsepower
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 392 entries, 0 to 397
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           392 non-null    float64
 1   cylinders     392 non-null    int64  
 2   displacement  392 non-null    float64
 3   horsepower    392 non-null    int64  
 4   weight        392 non-null    int64  
 5   acceleration  392 non-null    float64
 6   model year    392 non-null    int64  
dtypes: float64(3), int64(4)
memory usage: 24.5 KB


In [12]:
#Converting the Dataframe to Excel for Additional analysis in MS Excel
df_new.to_excel (r'C:\Users\mavje\Desktop\Auto-MPG.xlsx', index = False, header=True)